Library/KSCache.swift

```diff
@@ -0,0 +1,25 @@
+import Foundation
+
+public final class KSCache {
```

Comment:
this class replaces our `CacheProtocol` stuff. in swift 3,`NSCache` became generic over the key+values it holds, which is cool (!), but also kinda messed up the protocol stuff. Especially when it came to our subscript helpers because swift has no support for generic subscripts.

anyway, a much easier approach is just to wrap an instance of `NSCache` that is specialized to just `NSString` keys and `AnyObject` values, and then we can use it just as we used to.

/cc @stephencelis who i talked this through over slack...